### PR TITLE
[CELEBORN-168][FEATURE] Add disk usage related metrics for Worker

### PR DESCRIPTION
**Pull request by nafiyAix**
### What changes were proposed in this pull request?
Add disk usage related metrics for Worker.
BTW, shall we modify the PromServlet to support add customized label for Prom metrics, such as `device_type` or `mount_point` for disk related metrics?
And do we need to add metrics to monitor the hdfs capacity usage of each Worker, not sure if it is necessary.

### Why are the changes needed?
Better display current OS device's (not disk's) capacity of each Worker on monitor.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UTs

cc: @AngersZhuuuu

### CODE REVIEW

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Comment by waitinfuture** [comment]
> 

1. I think we can do that.
2. It's not of high priority. For shuffle read the client will directly read from hdfs, not through worker.

**Comment by waitinfuture** [comment]
Since we allow users to specify capacity that can be used by Celeborn, we might also consider that accordingly.
```
celeborn.worker.storage.dirs

"`dir1[:capacity=][:disktype=][:flushthread=],dir2[:capacity=][:disktype=][:flushthread=]`")
```

**Comment by waitinfuture** [comment]
> Since we allow users to specify capacity that can be used by Celeborn, we might also consider that accordingly.
> 
> ```
> celeborn.worker.storage.dirs
> 
> "`dir1[:capacity=][:disktype=][:flushthread=],dir2[:capacity=][:disktype=][:flushthread=]`")
> ```

We can add a new metric such as DeviceUsableCapacity which equals to min{freeSpace, configuredSpace - usedSpace}

**Comment by nafiyAix** [comment]
> > Since we allow users to specify capacity that can be used by Celeborn, we might also consider that accordingly.
> > ```
> > celeborn.worker.storage.dirs
> > 
> > "`dir1[:capacity=][:disktype=][:flushthread=],dir2[:capacity=][:disktype=][:flushthread=]`")
> > ```
> 
> We can add a new metric such as DeviceUsableCapacity which equals to min{freeSpace, configuredSpace - usedSpace}, see DiskInfo.configuredUsableSpace

+1, currently the new added metrics could only display the spaces status of OS capacity but not Celeborn capacity, will add it in this PR.

**Comment by AngersZhuuuu** [comment]
> > Since we allow users to specify capacity that can be used by Celeborn, we might also consider that accordingly.
> > ```
> > celeborn.worker.storage.dirs
> > 
> > "`dir1[:capacity=][:disktype=][:flushthread=],dir2[:capacity=][:disktype=][:flushthread=]`")
> > ```
> 
> We can add a new metric such as DeviceUsableCapacity which equals to min{freeSpace, configuredSpace - usedSpace}, see DiskInfo.configuredUsableSpace


capacity seems should be 
```
if (freeSpace < reservedSpace) {
    return 0
} else { 
  min{freeSpace - reservedSpace, configuredSpace - usedSpace}
}
```

cc @waitinfuture

**Comment by waitinfuture** [comment]
> 

I think the newest code is correct to DiskInfo's configuredUsableSpace and actualUsableSpace.

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/WorkerSource.scala [hunk 1]
```diff
@@ -91,4 +91,10 @@ object WorkerSource {
   val DiskBuffer = "DiskBuffer"
   val PausePushDataCount = "PausePushData"
   val PausePushDataAndReplicateCount = "PausePushDataAndReplicate"
+
+  // local device
+  val DeviceOSFreeCapacity = "DeviceOSFreeCapacity"
+  val DeviceOSTotalCapacity = "DeviceOSTotalCapacity"
+  val DeviceCelebornFreeCapacity = "DeviceCelebornFreeCapacity"
+  val DeviceCelebornTotalCapacity = "DeviceCelebornTotalCapacity"
```
Comment (AngersZhuuuu) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/WorkerSource.scala:99]: We should add a size unit in the name such as  `DeviceOSFreeCapacity(GB)` to indicate the size

**Pull request merged**